client/src/pages/Directory.tsx

```diff
@@ -0,0 +1,601 @@
+                dispatchSearch({
+                  ...searchState,
+                  type: SearchAction.UPDATE_QUERY,
+                  query: e.currentTarget.value,
+                })
+              }
+              icon="search"
+              loading={searchState.isLoading}
+            />
+          </div>
+          <div className="filters">
+            <h2>Filter by: </h2>
+            <Dropdown
+              className="custom-dropdown"
+              text="Roles"
+              options={roleOptions}
+              scrolling
+              clearable
+              selectOnBlur={false}
+              selectOnNavigation={false}
+              onChange={(e, { value }) => updateFilterKeys('role', `${value}`)}
+            />
+            <Dropdown
+              className="custom-dropdown"
+              text="Date Joined"
+              options={dateOptions}
+              scrolling
+              clearable
+              selectOnNavigation={false}
+              selectOnBlur={false}
+              onChange={(e, { value }) => updateFilterKeys('date', `${value}`)}
+            />
+            <Dropdown
+              className="custom-dropdown"
+              text="Topics of Interest"
+              options={interestOptions}
+              scrolling
+              multiple
+              clearable
+              selectOnNavigation={false}
+              selectOnBlur={false}
+              onChange={(e, { value }) =>
+                updateFilterKeys(
+                  'interests',
+                  parseSemanticMultiSelectTypes(value!),
+                )
+              }
+            />
+            <Dropdown
+              className="custom-dropdown"
+              text="Teams"
+              options={teamOptions}
+              scrolling
+              multiple
+              clearable
+              selectOnNavigation={false}
+              selectOnBlur={false}
+              onChange={(e, { value }) =>
+                updateFilterKeys('teams', parseSemanticMultiSelectTypes(value!))
+              }
+            />
+          </div>
+          <div className="directory">
+            {handleDirectoryFilter(searchState.users).map((user) => (
+              <Button
+                onClick={() => openContributorModal(user)}
+                key={user.oauthID}
+                className="result"
+              >
+                <img
+                  src={user.profilePic ? user.profilePic : DefaultProfile}
+                  alt="Profile"
+                  className="profile-picture"
+                />
+                <h2 className="text name">{`${user.firstName} ${user.lastName}`}</h2>
+                <h2 className="text">
+                  {/* Capitalize the first letter */}
+                  {user.role.slice(0, 1) + user.role.slice(1).toLowerCase()}
+                </h2>
+              </Button>
+            ))}
+          </div>
+        </div>
+      </div>
+    </>
+  );
+};
+
+export default Directory;
```

Comment:
A thought, but reading this file was heh a lot...I wonder if there's a way for you to split this into multiple files? I think it would be a lot cleaner that way

Reply:
+1 it wasn't even rendered by github by default lol

Reply:
Could you at least extract out the dropdowns & filtering functionality?